### PR TITLE
IIP-401: Allow to configure Tomcat SSL port number for HAC in Hybris panel

### PR DESCRIPTION
**Pull request by san4esmc**
Signed-off-by: Oleksandr Mishchuk san4es.mc@gmail.com

### CODE REVIEW

File: src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java [hunk 1]
```diff
@@ -43,10 +52,35 @@ public void createToolWindowContent(
         testConnectionButton.addActionListener(action->testConnection());
```
Comment (san4esmc) [inline src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java]: Small comment on lines 49-51: actionListener for JTextField will be only called when "Enter" key is pressed and it's unlikely that user will use "Enter" when changing information in text field. We, probably, need to change this similar as I did with generateUrlPreviewListener

**Comment by san4esmc** [comment]
Also I noticed that testing the connection is not working at least for me now. The request is done on correct URL, but certificate check fails. I saw in code that we already do some trust manager configuration to trust all certificates, but somehow it doesn't help.
Here is some stacktrace:
```
javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Certificates do not conform to algorithm constraints
	at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1959)
	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302)
	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296)
	at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1514)
	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)
	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1026)
	at sun.security.ssl.Handshaker.process_record(Handshaker.java:961)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1072)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1385)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1413)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1397)
	at sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:559)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:162)
	at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:652)
	at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:629)
	at org.jsoup.helper.HttpConnection.execute(HttpConnection.java:261)
	at com.intellij.idea.plugin.hybris.tools.remote.http.AbstractHybrisHacHttpClient.getResponseForUrl(AbstractHybrisHacHttpClient.java:186)
	at com.intellij.idea.plugin.hybris.tools.remote.http.AbstractHybrisHacHttpClient.getSessionId(AbstractHybrisHacHttpClient.java:162)
	at com.intellij.idea.plugin.hybris.tools.remote.http.AbstractHybrisHacHttpClient.login(AbstractHybrisHacHttpClient.java:63)
```

File: src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java (hunk 1 above)
Comment (zdary) [reply src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java]: by enter I meant entering data = typing, not the key "Enter". 
The idea was to have just one field for IP address and a port. i.e. "localhost:9002". The port part will be optional. if user doesn't enter port then it'll be taken from local properties dynamically.

the full url preview can a JLabel. JField could have key listener rendering the label every time new char is entered into the IPport field

**Comment by zdary** [comment]
Can you create a jira ticket regarding the SSL handshake, please?

File: src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java (hunk 1 above)
Comment (san4esmc) [reply src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java]: With my comment I meant that 
```
projectIpTextField.addActionListener(action->saveSettings());
```
will call `saveSettings()` ONLY when user press "Enter" key in the field. All other keys do not fire ActionListener in JEditText

**Comment by san4esmc** [comment]
Sure, will create ticket

File: src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java (hunk 1 above)
Comment (zdary) [reply src/com/intellij/idea/plugin/hybris/toolwindow/HybrisToolWindow.java]: ah, you are right. well spotted. We should change that.

**Pull request merged**